cmd/al-ctl/internal/al/discovery.go

```diff
@@ -28,7 +29,7 @@ func Discover(endpoint string, discoveryFile string) []*generated.DiscoverRespon
 		discoveryConfig, configError = config.NewDiscoveryConfigFromFile(discoveryFile)
 		if configError != nil {
 			log.Err(configError).Msg("Failed to read config file")
-			return nil
+			return nil, nil
```

Comment:
We should probably also return the error in this case.
```suggestion
			return nil, configError
```